File: app/src/main/java/com/ed/redditapp/ui/subreddit/SubredditDaggerModule.java
package com.ed.redditapp.ui.subreddit;

import com.ed.redditapp.ActivityScope;
import com.ed.redditapp.lib.api.RedditApi;
import com.ed.redditapp.ui.postlist.PostListAdapter;

import dagger.Module;
import dagger.Provides;

@Module
public class SubredditDaggerModule {
    private final SubredditActivity activity;

    public SubredditDaggerModule(SubredditActivity activity) {
        this.activity = activity;
    }

    @Provides
    @ActivityScope
    public SubredditModel provideModel(RedditApi redditApi) {
        return new SubredditModel(redditApi);
    }

    @Provides
    @ActivityScope
    public SubredditPresenter providePresenter(SubredditModel model) {
        return new SubredditPresenter(activity, model);
    }

    @Provides
    @ActivityScope
    public PostListAdapter provideAdapter() {
        return new PostListAdapter();
    }
}

File: app/src/main/java/com/ed/redditapp/ui/subreddit/SubredditDaggerModule.kt
package com.ed.redditapp.ui.subreddit

import com.ed.redditapp.ActivityScope
import com.ed.redditapp.lib.api.RedditApi
import com.ed.redditapp.ui.postlist.PostListAdapter
import dagger.Module
import dagger.Provides

@Module
class SubredditDaggerModule(private val activity: SubredditActivity) {
    @Provides @ActivityScope
    fun model(redditApi: RedditApi) = SubredditModel(redditApi)

    @Provides @ActivityScope
    fun presenter(model: SubredditModel) = SubredditPresenter(activity, model)

    @Provides @ActivityScope
    fun adapter() = PostListAdapter()
}